specs/MultiProfile.md

```diff
@@ -363,16 +384,36 @@ interface ICoreWebView2Profile2 : ICoreWebView2Profile {
   [propget] HRESULT CookieManager([out, retval] ICoreWebView2CookieManager** cookieManager);
 }
 
-[uuid(1c1ae2cc-d5c2-ffe3-d3e7-7857035d23b7), object, pointer_default(unique)]
-interface ICoreWebView2Profile3 : ICoreWebView2Profile2 {
-  /// All webviews on this profile will be closed, and the profile will be marked for deletion.
-  /// After the Delete() call completes, The render process of webviews on this profile will
-  /// asynchronously exit with the reason:`COREWEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED`.
-  /// See 'COREWEBVIEW2_PROCESS_FAILED_REASON::COREWEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED'
-  /// for more details. The profile directory on disk will be actually deleted when the browser
-  /// process exits. Webview2 creation will fail with the HRESULT is ERROR_INVALID_STATE(0x8007139FL)
-  /// if you create it with the same name as a profile that is being deleted.
-  HRESULT Delete(); 
+[uuid(2765B8BD-7C57-4B76-B8AA-1EC940FE92CC), object, pointer_default(unique)]
+interface ICoreWebView2StagingProfile4 : IUnknown {
+  /// After the API is called, the profile will be marked for deletion. The
+  /// local profile’s directory will be tried to delete at browser process
+  /// exit, if fail to delete, it will recursively try to delete at next
+  /// browser process start until successful.
+  /// The corresponding user's `ProfileDeleted` event handle function will
+  /// be triggered. After the function is triggered, continuing to use the
+  /// profile or its corresponding webviews is an undefined behavior.
+  /// For each WebView of the same profile, the WebView will be auto closed.
+  /// If the user has registered an event handler for ProfileDeleted event,
+  /// the event handle will be invoked before WebView is closed.
+  /// If create a new profile with the same name as the profile that has been
+  /// marked as deleted will be failure with the HRESULT:ERROR_INVALID_STATE
+  /// (0x8007139FL).
+  HRESULT Delete();
+}
+
+[uuid(cc39bea3-f6f8-471b-919f-fa253e2fff03), object, pointer_default(unique)]
+interface ICoreWebView2Staging9 : IUnknown {
```

Comment:
This event is about the profile so why don't we move it to the profile class? We should move the event to the CoreWebView2Profiler class and drop the 'Profile' prefix from the event name. Like `ICoreWebView2Profile4::add/remove_DeletionStarting` or in WinRT `CoreWebView2Profile.DeletionStarting`

Reply:
It is feasible to move the event to the CoreWebView2Profiler class, but the webview2 is more frequently used than profile, if we move the event to the CoreWebView2Profiler class, we should still close the webview2 after the user's handle function called and document that in the handle function, the user cannot use webview2 anymore, so it's better to put it directly in the webview2 class. And the profile can be regarded as an attribute of webview2 too, so, add an event to notify the webview2s that the profile is deleting is appropriate.

Reply:
If we're informing the end developer that the profile is getting deleted that should be an event on the profile. If we're expecting the end developer to only care about it because it means the CoreWebView2 is closing then perhaps we want a CoreWebView2.Closed event instead or in addition to CoreWebView2Profile.DeletionStarting event

Reply:
ok has change CoreWebview2.ProfileDelete -> CoreWebView2.Closed.